models/migrations/v1_21/v279.go

```diff
@@ -0,0 +1,16 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package v1_21 //nolint
```

Comment:
Now we are in 1.22 dev cycle could you move this to the new package?

Reply:
Would that prevent a backport?

Reply:
I also think it would be better to backport it to 1.21 instead of treating it as a 1.22 feature.

Reply:
Since we have created v1.21 branch, we should not do that.

Reply:
This PR is made to be backported to 1.21, as it improves the performance and does not introduce new features

Reply:
I can understand this, maybe it's a special case and this will take much time for a big instance.

Reply:
I also am in favour of backporting

Reply:
I guess we have to backport such migrations in the v1.22 folder into the v1.21 branch, because if it's done the other way, it would force a re-numbering of future migrations which would break the migration system.

Generally, I think it was a bad idea to split the migrations into folders. Ideally I'd like to see migrations identified by strings instead of numbers, much less merge conflicts that way as well.

Reply:
But then it will have hundreds of files in the folder?

Reply:
could split by year instead of version if number of files is a concern, e.g. `2023`, `2024` etc.

Reply:
Or simply commit it in the release it is backported to, just as @JakobDev did?
I don't quite see the problem here…

Reply:
We don't have any 1.22 migration yet, so I don't see the problem.

Reply:
@techknowlogick @lunny any objections/other ideas?

Reply:
I agree with @silverwind that we need to update how we do migrations, preferably tracking all migrations that are done instead of a counter (I'm a fan of how rails does their migrations). I wrote a library for exactly this for xorm, https://github.com/techknowlogick/xormigrate which is used by a few projects, and does allow for splitting things out into directories to ensure 100s of files aren't all in the same directory. This however is a larger conversation than just this PR.

Comment scoped to this PR: My personal preference would be to change it to 1.22 as that's the dev cycle we are in, and then backport it with the same 1.22 directory. However we are in the stage where migrations can still be backported, and 1.21 is not fully released yet, so I can see using 1.21 still.

Reply:
I think the directory should not represent the development circle. It should represent for which version the migration is.

Reply:
ultimately it won't matter if the 1.21 has files in the 1.22 folder. so let's go with that.